electron-app/src/server/websites/pixelfed/pixelfed.service.ts

```diff
@@ -180,85 +195,69 @@ export class Pixelfed extends Website {
+    let statusOptions: any = {
+      sensitive: isSensitive,
+      visibility: options.visibility || 'public',
+      in_reply_to_id: lastId,  
+      spoiler_text: "",    
+    }  
+    let status = "";
+
     for (let i = 0; i < chunks.length; i++) {
-      let form = undefined;
       if (i === 0) {
-        form = {
-          status: `${options.useTitle && data.title ? `${data.title}\n` : ''}${
-            data.description
-          }`.substring(0, maxChars),
-          sensitive: isSensitive,
-          visibility: options.visibility || 'public',
-          media_ids: chunks[i].map((media) => media.id),
-        };
-      } else {
-        form = {
-          sensitive: isSensitive,
-          visibility: options.visibility || 'public',
-          media_ids: chunks[i].map((media) => media.id),
-          in_reply_to_id: lastId,
-        };
-      }
+        status = `${options.useTitle && data.title ? `${data.title}\n` : ''}${
+          data.description
+          }`.substring(0, maxChars);
+        statusOptions.media_ids = chunks[i].map((media) => media.id);
+     }
 
-      const tags = this.formatTags(data.tags);
+      this.logger.debug(`Number of tags set ${data.tags.length}`);
 
-      // Update the post content with the Tags if any are specified - for Pixelfed, we need to append
+      // Update the post content with the Tags if any are specified - for Pixelfed, we need to append 
       // these onto the post, *IF* there is character count available.
-      if (tags.length > 0) {
-        form.status += '\n\n';
+      if (data.tags.length > 0) {
+        status += "\n\n";
       }
 
-      tags.forEach((tag) => {
-        let remain = maxChars - form.status.length;
+      data.tags.forEach(tag => {
+        let remain = maxChars - status.length;
         let tagToInsert = tag;
-        if (remain > tagToInsert.length) {
-          form.status += ` ${tagToInsert}`;
+        if (!tag.startsWith('#')) {
+          tagToInsert = `#${tagToInsert}`
+        }
+        if (remain > (tagToInsert.length)) {
+          status += ` ${tagToInsert}`
         }
         // We don't exit the loop, so we can cram in every possible tag, even if there are short ones!
-      });
-
+      })
+      
       if (options.spoilerText) {
-        form.spoiler_text = options.spoilerText;
+        statusOptions.spoiler_text = options.spoilerText;
       }
 
-      const post = await M.post('statuses', form);
-      lastId = post.data.id;
-
-      if (!lastId || post.data.error) {
+      M.postStatus(status, statusOptions).then((result) => {
+        lastId = result.data.id;
+        let res = result.data as Entity.Status;
+        return this.createPostResponse({ source: res.url });
+      }).catch((err: Error) => {
         return Promise.reject(
-          this.createPostResponse({ message: post.data.error, additionalInfo: post.data }),
+          this.createPostResponse({ message: err.message }),
         );
-      }
+      })
```

Comment:
Needs to return or await.
Looks like this just falls through.

Reply:
Resolved